### PR TITLE
NIFI-9812 Skip Native Lib loading tests on ARM64

### PR DESCRIPTION
**Pull request by kevdoran**
- Migrate tests in nifi-framework-nar-loading-utils to JUnit5
- Annotate tests that use x86_64 native binaries to be conditional on x86_64 os.arch

# Summary

[NIFI-9812](https://issues.apache.org/jira/browse/NIFI-9812)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by ottobackwards** [comment]
LGTM +1

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the improvement @kevdoran! The update to JUnit 5 looks good.

The only additional adjustment necessary is to remove the `assumeTrue()` calls that check the operating system. These calls are no longer necessary when using the `EnabledOnOs` and `EnabledIfSystemProperty` annotations.

**Comment by kevdoran** [comment]
Yep, I agree. Will remove the assumptions

**Review by exceptionfactory — APPROVED** [review]
Thanks for removing the `assume` calls @kevdoran, looks good! +1 merging

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-nar-loading-utils/src/test/java/org/apache/nifi/nar/TestNarLoader.java [hunk 1]
```diff
@@ -33,20 +33,18 @@
...
 import java.util.List;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
+
+@EnabledOnOs({ OS.MAC, OS.LINUX })
+@EnabledIfSystemProperty(named = "os.arch", matches = "x86_64")
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-nar-loading-utils/src/test/java/org/apache/nifi/nar/TestNarLoader.java:41]: On further review, it seems better to use `DisabledOnOs` with `OS.WINDOWS` and use `DisabledIfSystemProperty` matching `aarch64`.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Following the pattern of the other builds, further evaluation indicates that the `TestNarLoader` should still run on `am64`, but would not based on limiting the architecture to `x86_64`.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @kevdoran, the latest version looks good! +1 merging

**Pull request closed without merging**